engine/src/main.cpp

```diff
@@ -114,6 +114,14 @@ int main(int argc, char** argv) {
   app.add_flag("--interpolate-undefined", stack_options.interpolate_undefined,
                "Interpolate undefined variables with empty strings");
 
+  // The --strict flag here is useful for all our smoketests, since this is the
+  // combination of flags we use for maximum reproducibility / isolation.
+  // Note: This option also affects / overwrites options for the run subcommand!
+  app.add_flag("-t,--strict", stack_options.strict_mode,
```

Comment:
If anything, `--test-mode`, but I feel like that is also not awefully clear. Users might wonder what exactly we are testing... cloe-engine itself?

If not `--strict`, how about `--restrict` or `--restricted`. The environment variable would be `CLOE_RESTRICTED_MODE`.

Thoughts?

Reply:
We could also call it ``--pedantic``, similar to the compiler options.
Pedantic because:
- Plugin inclusion needs to be specified explicitely.
- Configuration needs to be specified explicitely.
- Require simulation success or we return non-null exit code.